### PR TITLE
Make docker collector threadsafe

### PR DESCRIPTION
**Pull request by gnufied**

### CODE REVIEW

File: src/fullerite/collector/docker_stats.go [hunk 1]
```diff
@@ -159,6 +158,16 @@ func (d DockerStats) getDockerContainerInfo(container *docker.Container) {
 	}
 }
 
+func (d *DockerStats) extractMetrics(container *docker.Container, stats *docker.Stats) []metric.Metric {
+	defer d.mu.Unlock()
```
Comment (venkey-ariv) [inline src/fullerite/collector/docker_stats.go]: Shouldn't this be after acquiring the lock?

Comment (gnufied) [inline src/fullerite/collector/docker_stats.go]: good point.

**Comment by tiras-j** [comment]
+1 lgtm.

To solidify my thinking though, the reason we need the synchronization is because multiple goroutines could be accessing the d.previousCPUvalues map at the same time. And even though they're accessing different keys (container.ID is of course unique to each container) maps themselves are not thread safe in go so we need synchronization around it. That sound right?

**Comment by poros** [comment]
+1

**Pull request merged**